python/hail/dataset.py

```diff
@@ -4980,38 +4987,98 @@ def tdt(self, pedigree, root='va.tdt'):
 
         :py:meth:`~hail.VariantDataset.tdt` only considers complete trios (two parents and a proband) with defined sex.
 
-        PAR is currently defined with respect to reference `GRCh37 <http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/human/>`__:
+        PAR is currently defined with respect to reference
+        `GRCh37 <http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/human/>`__:
 
         - X: 60001-2699520
         - X: 154931044-155260560
         - Y: 10001-2649520
         - Y: 59034050-59363566
 
-        :py:meth:`~hail.VariantDataset.tdt` assumes all contigs apart from X and Y are fully autosomal; decoys, etc. are not given special treatment.
-
-        **Annotations**
+        :py:meth:`~hail.VariantDataset.tdt` produces a key table with the following columns:
 
-        :py:meth:`~hail.VariantDataset.tdt` adds the following annotations:
+         - **v** (*Variant*) -- Variant tested.
 
-         - **tdt.nTransmitted** (*Int*) -- Number of transmitted alternate alleles.
+         - **nTransmitted** (*Int32*) -- Number of transmitted alternate alleles.
 
-         - **va.tdt.nUntransmitted** (*Int*) -- Number of untransmitted alternate alleles.
+         - **nUntransmitted** (*Int32*) -- Number of untransmitted alternate alleles.
 
-         - **va.tdt.chi2** (*Double*) -- TDT statistic.
+         - **chi2** (*Float64*) -- TDT statistic.
 
-         - **va.tdt.pval** (*Double*) -- p-value.
+         - **pval** (*Float64*) -- p-value.
 
         :param pedigree: Sample pedigree.
         :type pedigree: :class:`~hail.representation.Pedigree`
 
-        :param root: Variant annotation root to store TDT result.
-
-        :return: Variant dataset with TDT association results added to variant annotations.
-        :rtype: :py:class:`.VariantDataset`
+        :return: Key table with TDT association results.
+        :rtype: :py:class:`.KeyTable`
         """
 
-        jvds = self._jvdf.tdt(pedigree._jrep, root)
-        return VariantDataset(self.hc, jvds)
+        trio_matrix = self.trio_matrix(pedigree, complete_trios=True)
+        trio_matrix = trio_matrix.filter_samples_expr('isDefined(sa.isFemale)')
+
+        hom_ref = 0
+        het = 1
+        hom_var = 2
+
+        auto = 2
+        hemiX = 1
+
+        t = TDict(TStruct(['kid', 'dad', 'mom', 'ploidy'], [TInt32(), TInt32(), TInt32(), TInt32()]), TArray(TInt32()))
```

Comment:
you mentioned that 'ploidy' might not be the right term here; I think the original uses 'copyState'?